File: IslandPerimeter.c
#include <stdio.h>

//Author:liuiuge(1220595883@qq.com)
int islandPerimeter(int** grid, int gridRowSize, int gridColSize) {
    int cnt = 0, neighbor = 0;
    for(int i=0;i<gridRowSize; ++i)
        for(int j=0; j<gridColSize; ++j)
        {
            if(grid[i][j])
            {
                cnt++;
                if(i>0&&grid[i-1][j])
                    neighbor++;
                if(j>0&&grid[i][j-1])
                    neighbor++;
            }

        }
    return 4*cnt-2*neighbor;
}

File: IslandPerimeter.py
#!/usr/bin/env python
# coding=utf-8
class Solution(object):
    def islandPerimeter(self, grid):
        m,n = len(grid),len(grid[0])
        cnt,neighbor=0,0
        for i in xrange(m):
            for j in xrange(n):
                if(grid[i][j]):
                    cnt+=1
                    if i<m-1 and grid[i+1][j]:
                         neighbor+=1
                    if j<n-1 and grid[i][j+1]:
                         neighbor+=1
        return 4*cnt-2*neighbor
